### DISCUSSION TITLE
Initial connection timeout or reconnect when namespace is not yet created

### DISCUSSION THREAD

**Discussion by andredewaard** [socketio/socket.io#4243] [Q&A]
Is it possible to open a connection only after a certain time?
So an initial connection timeout.

We create different namespaces for each 'object' inside the database. When something happens on a certain 'object' when broadcast an event on that specific namespace. 
When we create an 'object' the namespace for this object is not created immediately so when trying to connect to the websocket the connection to the server is there only the namespace doesnt exist yet. SocketIO then throws an error instead of trying to reconnect.

**Comment by darrachequesne** [comment]
Hi! I'm not sure to totally understand your use case, but you should be able to use the `autoConnect` option:

```js
import { io } from "socket.io-client";

const socket = io({
  autoConnect: false
});

// and then later
socket.connect();

// or after a given delay
setTimeout(() => {
  socket.connect();
}, 5000);
```

Reference: https://socket.io/docs/v4/client-options/#autoconnect
